executor/src/witgen/jit/debug_formatter.rs

```diff
@@ -0,0 +1,356 @@
+                self.format_unary_operation(op, row_offset, simplified)
+            }
+        };
+        pad_center([name, value, rc])
+    }
+
+    fn format_binary_operation(
+        &self,
+        op: &AlgebraicBinaryOperation<T>,
+        row_offset: i32,
+        simplified: bool,
+    ) -> [String; 3] {
+        // TODO precedence
+
+        if simplified {
+            if let Some(s) = self.try_format_binary_operation_simplified(op, row_offset) {
+                return s;
+            }
+        }
+        let [left, right] = [
+            self.format_expression(&op.left, row_offset, simplified),
+            self.format_expression(&op.right, row_offset, simplified),
+        ];
+
+        left.into_iter()
+            .zip(right)
+            .enumerate()
+            .map(|(i, (l, r))| {
+                if i == 0 {
+                    format!("({l} {} {r})", op.op)
+                } else {
+                    format!(" {l}   {r} ")
+                }
+            })
+            .collect_vec()
+            .try_into()
+            .unwrap()
+    }
+
+    fn try_format_binary_operation_simplified(
+        &self,
+        op: &AlgebraicBinaryOperation<T>,
+        row_offset: i32,
+    ) -> Option<[String; 3]> {
+        let left = self.try_to_known(&op.left, row_offset);
+        let right = self.try_to_known(&op.right, row_offset);
+
+        match op.op {
+            AlgebraicBinaryOperator::Add => {
+                if left.map(|v| v == 0.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.right, row_offset, true))
+                } else if right.map(|v| v == 0.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.left, row_offset, true))
+                } else {
+                    None
+                }
+            }
+            AlgebraicBinaryOperator::Sub => {
+                if left.map(|v| v == 0.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.right, row_offset, true))
+                } else if right.map(|v| v == 0.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.left, row_offset, true))
+                } else {
+                    None
+                }
+            }
+            AlgebraicBinaryOperator::Mul => {
+                if left.map(|v| v == 1.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.right, row_offset, true))
+                } else if right.map(|v| v == 1.into()).unwrap_or(false) {
+                    Some(self.format_expression(&op.left, row_offset, true))
+                } else {
+                    None
+                }
+            }
+            AlgebraicBinaryOperator::Pow => None,
+        }
+    }
+
+    fn format_unary_operation(
+        &self,
+        op: &AlgebraicUnaryOperation<T>,
+        row_offset: i32,
+        simplified: bool,
+    ) -> [String; 3] {
+        // TODO precedence
```

Comment:
now or later? (also about the other TODO's)

Reply:
Later :)